src/main/scala/no/ndla/imageapi/controller/NdlaController.scala

```diff
@@ -38,19 +39,59 @@ abstract class NdlaController extends ScalatraServlet with NativeJsonSupport wit
   }
 
   error {
-    case v: ValidationException => halt(status = 400, body = Error(Error.VALIDATION, v.getMessage))
-    case e: IndexNotFoundException => halt(status = 500, body = Error.IndexMissingError)
+    case v: ValidationException => BadRequest(Error(Error.VALIDATION, v.getMessage))
+    case e: IndexNotFoundException => InternalServerError(Error.IndexMissingError)
+    case i: ImageNotFoundException => NotFound(Error(Error.NOT_FOUND, i.getMessage))
     case t: Throwable => {
       logger.error(Error.GenericError.toString, t)
       halt(status = 500, body = Error.GenericError)
     }
   }
 
+  private val streamRenderer: RenderPipeline = {
+    case f: ImageStream =>
+      contentType = f.contentType
+      org.scalatra.util.io.copy(f.stream, response.getOutputStream)
+  }
+
+  override def renderPipeline = streamRenderer orElse super.renderPipeline
+
+  def isNumber(value: String): Boolean = {
+    value.forall(_.isDigit)
+  }
+
   def long(paramName: String)(implicit request: HttpServletRequest): Long = {
     val paramValue = params(paramName)
-    paramValue.forall(_.isDigit) match {
+    isNumber(paramValue) match {
```

Comment:
Jeg får warnings i IntelliJ på sånne enkle matchinger nå for tiden, om at den bør omgjøres til en if.
I dette tilfellet så kunne vel hele funksjonen blitt forenklet litt:

```
def long(....) = {
   val paramValue = params(paramName)
   if(!isNumber(paramValue) 
      throw new ValidationException(...)

   paramValue.toLong
}
```